avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/ArrayVisitor.java

```diff
@@ -35,8 +37,8 @@ public Schema builtAvroSchema() {
             throw new IllegalStateException("No element schema created for: "+_type);
         }
         Schema schema = Schema.createArray(_elementSchema);
-        if (_type.isArrayType()) {
-            schema.addProp(AVRO_SCHEMA_PROP_CLASS, _type.toCanonical());
+        if (!_type.hasRawClass(List.class)) {
+            schema.addProp(AVRO_SCHEMA_PROP_CLASS, AvroSchemaHelper.getTypeId(_type));
```

Comment:
@baharclerode True, but keep in mind that Avro RI encoder wouldn't be seeing original value at all: `AvroGenerator` will essentially re-build contents that were split by `jackson-databind`. So encoder need not be aware of any of that. It will be just passed a `List`. However, type of the POJO property may be something completely different.

I don't have enough context here to know for sure; I guess I am ok with this check for now, and it can be improved upon if and as need be.